### PR TITLE
Add RemoveSubscriber method to SubscriberRepository & unit tests

### PR DESCRIPTION
**Pull request by tabman83**

### CODE REVIEW

File: src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs [hunk 1]
```diff
@@ -104,6 +104,26 @@ public Task<OperationResult<SubscriberEntity>> UpdateSubscriberAsync(SubscriberE
             return UpdateSubscriberInternalAsync(subscriberEntity);
         }
 
+        public Task<OperationResult<SubscriberEntity>> RemoveSubscriberAsync(SubscriberEntity subscriberEntity)
```
Comment (chomick) [inline src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs:107]: Do we need entire entity or can we just pass the minimum, which is the identifier?

Comment (tabman83) [reply src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs:107]: Needed to construct the partition key

**Review by chomick — APPROVED** [review]
I would reduce the contract for delete to mini. i.e. identifier (that should give info event name as well, i guess?)

File: src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs (hunk 1 above)
Comment (tabman83) [reply src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs:107]: Other option would be to accept subscriberId and eventName

Comment (chomick) [reply src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs:107]: is subscriber id (the object - do we still have it?) not built of event & subscriber names?

Comment (tabman83) [reply src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs:107]: Yes can use SubscriberId.EventName !! - SubscriberId would be enough then

Comment (chomick) [reply src/CaptainHook.Storage.Cosmos/SubscriberRepository.cs:107]: if not, we may need to pass id & event name.

**Review by dawidkeshop — APPROVED** [review]

**Comment by nikhilgirraj** [comment]
Changed the parameter to RemoveSubscriber.

**Pull request merged**